Add E2E tests for DAG Runs page

lin121291 opened the pull request:
## What change does this PR introduce?

Adds end-to-end (E2E) tests for the DAG runs page (`/dag_runs`) to improve test coverage of the UI.

## What does this PR do?

### New Files:
- `airflow-core/src/airflow/ui/tests/e2e/pages/DagRunsPage.ts` - Page Object Model for DAG runs page
- `airflow-core/src/airflow/ui/tests/e2e/specs/dag-runs.spec.ts` - Test specifications

### Test Coverage (as per DAGRUNS-001):
- [x] **Navigate to DAG Runs page** - Verifies page loads and table renders
- [x] **Verify run details display** - Checks DAG ID, Run ID, State, Start Time, End Time
- [x] **Verify state visual distinction** - Validates different states (success, failed, running) have distinct badge colors
- [x] **Verify filtering works** - Ensures filter functionality is accessible and functional

## Related Issues

Closes #59309

lin121291 committed:
Add E2E tests for DAG Runs page

- Add DagRunsPage page object with locators and actions
- Add test specs for DAG runs list, filtering, and navigation
- Cover DAG run state changes

lin121291 committed:
Merge branch 'main' into ui/e2e-dag-runs-tests

vatsrahul1001 committed:
Merge branch 'main' into ui/e2e-dag-runs-tests

lin121291 committed:
Merge branch 'main' into ui/e2e-dag-runs-tests

vatsrahul1001 committed:
Merge branch 'main' into ui/e2e-dag-runs-tests

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/pages/DagRunsPage.ts:
```diff
@@ -0,0 +1,173 @@
+    await stateFilterOption.click();
+    await this.page.waitForTimeout(500);
+
+    await expect(this.dagRunsTable).toBeVisible();
```
We are not verifying filtering results here. We should check that the filter works

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/specs/dag-runs.spec.ts:
```diff
@@ -0,0 +1,48 @@
+    await dagRunsPage.verifyRunDetailsDisplay();
+  });
+
+  test("should display different run states with visual badges", async () => {
```
Let's remove this test and we can cover states check in filter tests

vatsrahul1001 commented on airflow-core/src/airflow/ui/tests/e2e/specs/dag-runs.spec.ts:
```diff
@@ -0,0 +1,48 @@
+    await dagRunsPage.verifyStateVisualDistinction();
+  });
+
+  test("should have filtering functionality", async () => {
```
As mentioned [here](https://github.com/apache/airflow/pull/59919#issuecomment-3713306224). We should filter and check whether filter is applied correctly or not. Let's filter with dag_id(Create runs for different dag_id in beforeAll) and state(create run with failed state in before all )

lin121291 committed:
feat: enhance DAG runs E2E tests with proper filtering and pagination verification

- Set 60s timeout for WebKit compatibility
- Add beforeAll hook to create test data via API (failed/success states, multiple DAG IDs)
- Replace waitForTimeout with proper element waits
- Verify state and DAG ID filters return correct results
- Add pagination test with offset/limit parameters
- Remove redundant visual badge test

lin121291 committed:
Merge branch 'main' into ui/e2e-dag-runs-tests

vatsrahul1001 committed:
Merge branch 'main' of github.com:astronomer/airflow into fix/pr-59919-improvements

vatsrahul1001 committed:
fix(ui/e2e): improve DAG runs page tests reliability

vatsrahul1001 committed:
Merge branch 'main' into ui/e2e-dag-runs-tests
